HEATDamage.cs

```diff
@@ -0,0 +1,80 @@
+using System;
+using System.Text;
+using System.Diagnostics;
+
+namespace PhysicalDamage.Core
+{
+    class HeatDamage:Damage
+    {
+        public struct HeatData
+        {
+            /// <summary>
+            /// The pierce of the shot
+            /// </summary>
+            public float Pierce;
+            /// <summary>
+            /// Multiplier for armor degradation
+            /// </summary>
+            public float Degradation;
+            /// <summary>
+            /// Multiplier for health damage
+            /// </summary>
+            public float HealthDamageFactor;
+        }
+
+        private HeatData _heatData;
+
+        public finalState(HeatData data,Target target)
+            : base(DamageTypes.HEAT, target)
+        {
+            this._heatData = data;
+        }
+
+        public override Target CalculateDamage()
+        {
+            Target finalState = this.CurrentTarget;
+            HeatData he = _heatData;
+
+            // No air friction attenuation as HEAT round detonates on surface of the armor
+
+            if (finalState.EraData.Value > 0.0f) {
+                // Calculate effects of ERA
+                float finalEra = Math.Max(
+                    0.0f,
+                    finalState.EraData.Value - he.Pierce * finalState.EraData.HeatFractionMultiplier
+                );
+                finalState.EraData.Value = finalEra;
+                
+                he.Pierce = CalculatePostERAPierce(
+                    he.Pierce,
+                    finalState.EraData.HeatFractionMultiplier
+                );
+            }
+
+            // Armor degradation
+            float finalArmor = Math.Max(
```

Comment:
Instead of Math.Max, I'm pretty certain you could use a [? (ternary)](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/conditional-operator) operator in this context.